src/coprocessor/dag/expr/compare.rs

```diff
@@ -161,6 +161,70 @@ impl FnCall {
         do_coalesce(self, |v| v.eval_json(ctx, row))
     }
 
+    pub fn greatest_int(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<i64>> {
+        do_get_extremum(self, |e| e.eval_int(ctx, row), max)
+    }
+
+    pub fn greatest_real(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<f64>> {
+        do_get_extremum(self, |e| e.eval_real(ctx, row), |x, y| x.max(y))
+    }
+
+    pub fn greatest_decimal<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, Decimal>>> {
+        do_get_extremum(self, |e| e.eval_decimal(ctx, row), max)
+    }
+
+    pub fn greatest_time<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, Time>>> {
+        do_get_extremum(self, |e| e.eval_time(ctx, row), max)
+    }
+
+    pub fn greatest_string<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        do_get_extremum(self, |e| e.eval_string(ctx, row), max)
+    }
+
+    pub fn least_int(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<i64>> {
+        do_get_extremum(self, |e| e.eval_int(ctx, row), min)
+    }
+
+    pub fn least_real(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<f64>> {
+        do_get_extremum(self, |e| e.eval_real(ctx, row), |x, y| x.min(y))
+    }
+
+    pub fn least_decimal<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, Decimal>>> {
+        do_get_extremum(self, |e| e.eval_decimal(ctx, row), min)
+    }
+
+    pub fn least_time<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, Time>>> {
```

Comment:
ditto, https://github.com/pingcap/tidb/blob/master/expression/builtin_compare.go#L746

Reply:
ooh, I was too careless to notice this